### PR TITLE
Add dra-artifact pipeline placeholder

### PR DESCRIPTION
**Pull request by marc-gr**
## What does this PR do?

Add the placeholder for the dra pipeline in order to be able to publish it to buildkite and test https://github.com/elastic/elastic-agent/pull/2812

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -104,3 +104,48 @@ spec:
...
+    spec:
+      pipeline_file: ".buildkite/pipeline.dra-artifacts.yml"
+      provider_settings:
+        build_branches: true
+        build_pull_request_forks: false
+        build_pull_requests: true
+        publish_commit_status: true
+        publish_commit_status_per_step: false
+      repository: elastic/elastic-agent
+      schedules:
```
Comment (alexsapran) [inline catalog-info.yaml:138]: should we put this after we get this working?

File: catalog-info.yaml [hunk 2]
```diff
@@ -104,3 +104,48 @@ spec:
...
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: dra-artifacts
+      description: Buildkite pipeline for packaging Elastic Agent core binary and publish it to DRA
+    spec:
+      pipeline_file: ".buildkite/pipeline.dra-artifacts.yml"
+      provider_settings:
```
Comment (alexsapran) [inline catalog-info.yaml:131]: Let's put the same `provider_settings` as the other pipelines to make sure builds are only triggered by the `elasticmachine`

File: catalog-info.yaml [hunk 3]
```diff
@@ -104,3 +104,48 @@ spec:
...
+
+spec:
+  type: buildkite-pipeline
+  owner: group:ingest-fp
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: dra-artifacts
```
Comment (alexsapran) [inline catalog-info.yaml:127]: should we name it something more specific, like `elastic-agent-binary-dra`

**Review by alexsapran — APPROVED** [review]

**Review by pchila — APPROVED** [review]
LGTM

**Pull request merged**